ic-factory/src/top_up.rs

```diff
@@ -0,0 +1,151 @@
+use candid::Principal;
+
+use ic_canister::{call_virtual_responder, virtual_canister_call};
+use ic_exports::{
+    cycles_minting_canister::{
+        IcpXdrConversionRateCertifiedResponse, NotifyError, NotifyTopUp, DEFAULT_CYCLES_PER_XDR,
+        MEMO_TOP_UP_CANISTER,
+    },
+    ic_base_types::{CanisterId, PrincipalId},
+    ledger_canister::{
+        AccountIdentifier, BlockHeight, Subaccount, Tokens, TransferArgs, DEFAULT_TRANSFER_FEE,
+        TOKEN_SUBDIVIDABLE_BY,
+    },
+};
+
+use crate::error::FactoryError;
+
+/// This Principal is a slice equivalent to `rkp4c-7iaaa-aaaaa-aaaca-cai`.
+const CYCLES_MINTING_CANISTER: Principal = Principal::from_slice(&[0, 0, 0, 0, 0, 0, 0, 4, 1, 1]);
```

Comment:
We still use a default one? We need to be able to parametrise this principal as we do with ledger in our repos. Could be a separate PR though

Reply:
We don't actually need to parametrise this, because we will be using the NNS cmc canister to communicate directly hence no need for it.